File: src/main/java/io/github/phantamanta44/libnine/client/event/ClientTickHandler.java
package io.github.phantamanta44.libnine.client.event;

import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.common.gameevent.TickEvent;

import java.util.ArrayList;
import java.util.Collection;

public class ClientTickHandler {

    private static final Collection<Listener> listeners = new ArrayList<>();
    private static long tick;

    public static long getTick() {
        return tick;
    }

    @SubscribeEvent
    public void onTick(TickEvent.ClientTickEvent event) {
        if (event.phase == TickEvent.ClientTickEvent.Phase.END) {
            ++tick;
            for (Listener listener : listeners) {
                listener.onClientTick(tick);
            }
        }
    }

    public void registerListener(Listener listener) {
        listeners.add(listener);
    }

    public interface Listener {

        void onClientTick(long tick);

    }

}

File: src/main/kotlin/xyz/phanta/libnine/client/event/ClientTickHandler.kt
package xyz.phanta.libnine.client.event

import net.minecraftforge.event.TickEvent
import java.util.function.Consumer

class ClientTickHandler : Consumer<TickEvent.ClientTickEvent> {

    var tick: Long = 0L
        private set

    private val listeners: MutableList<Listener> = mutableListOf()

    fun registerListener(listener: Listener) {
        listeners += listener
    }

    override fun accept(event: TickEvent.ClientTickEvent) {
        if (event.phase == TickEvent.Phase.END) {
            ++tick
            listeners.forEach { it.onClientTick(tick) }
        }
    }

    interface Listener {

        fun onClientTick(tick: Long)

    }

}
